contracts/queue/src/contract.rs

```diff
@@ -0,0 +1,202 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+
+use cosmwasm_std::{
+    from_slice, to_vec, Api, Binary, Env, Extern, Order, ParseErr, Response, Result, SerializeErr,
+    Storage,
+};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InitMsg {}
+
+// we store one entry for each item in the queue
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct State {
+    pub value: i32,
+}
+
+// failure modes to help test wasmd, based on this comment
+// https://github.com/cosmwasm/wasmd/issues/8#issuecomment-576146751
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum HandleMsg {
+    // Push will add some value to the end of list
+    Push { value: i32 },
+    // Pop will remove value from start of the list
+    Pop {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum QueryMsg {
+    // how many items are in the queue
+    Count {},
+    // total of all values in the queue
+    Sum {},
+    //    // first element in the queue
+    //    First {},
+    //    // last element in the queue
+    //    Last {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct CountResponse {
+    pub count: i32,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct SumResponse {
+    pub sum: i32,
+}
+
+// init is a no-op, just empty data
+pub fn init<S: Storage, A: Api>(
+    _deps: &mut Extern<S, A>,
+    _env: Env,
+    _msg: InitMsg,
+) -> Result<Response> {
+    Ok(Response::default())
+}
+
+pub fn handle<S: Storage, A: Api>(
+    deps: &mut Extern<S, A>,
+    env: Env,
+    msg: HandleMsg,
+) -> Result<Response> {
+    match msg {
+        HandleMsg::Push { value } => do_push(deps, env, value),
+        HandleMsg::Pop {} => do_pop(deps, env),
+    }
+}
+
+const FIRST_KEY: u8 = 1;
+
+fn do_push<S: Storage, A: Api>(deps: &mut Extern<S, A>, _env: Env, value: i32) -> Result<Response> {
+    // find the last element in the queue and extract key
+    let last = deps
```

Comment:
`last` -> `last_key` as in contrast to the next function, you only store the key